src/alerts/check_signals_updates.R

```diff
@@ -0,0 +1,207 @@
+#' Requires workflows to be named following this convention: `monitor_<indicator_id>.yaml`
+#'
+#' NOTE: More data will be returned as each workflow is run more times. Perhaps won't scale well,
+#' depending on how many runs a workflow will have
+#'
+#' @param indicator_id ID of the indicator
+#'
+#' @returns JSON API response
+gh_status <- function(indicator_id) {
+  # TODO Temp cleaning before workflows are renamed to match indicator ids
+  ind <- sub("^[^_]*_", "", indicator_id)
+  workflow_id <- paste0("monitor_", ind, ".yaml")
+  token <- get_env("GH_TOKEN")
+  url <- paste0("https://api.github.com/repos/ocha-dap/hdx-signals/actions/workflows/", workflow_id, "/runs")
+  response <- GET(url, add_headers(Authorization = paste("token", token)))
+  return(response)
+}
+
+#' Builds and posts a message to Slack, using incoming webhooks
+#' See API docs: https://api.slack.com/messaging/webhooks
+#'
+#' @param header_text Text for the message header
+#' @param status_text Text for the GitHub actions status report
+#' @param signals_text Text for the signals status report
+#'
+#' @returns Nothing. Message is posted to slack and will log an error if not successful
+slack_post_message <- function(header_text, status_text, signals_text) {
+  # See https://app.slack.com/block-kit-builder for prototyping layouts in JSON
+  msg <- list(
+    blocks = list(
+      list(
+        type = "section",
+        text = list(
+          type = "plain_text",
+          text = header_text,
+          emoji = TRUE
+        )
+      ),
+      list(
+        type = "section",
+        text = list(
+          type = "mrkdwn",
+          text = signals_text
+        )
+      ),
+      list(type = "divider"),
+      list(
+        type = "context",
+        elements = list(
+          list(
+            type = "mrkdwn",
+            text = status_text
+          )
+        )
+      ),
+      list(type = "divider")
+    )
+  )
+  json_body <- toJSON(msg, pretty = TRUE, auto_unbox = TRUE)
+  response <- POST(get_env("SLACK_URL"), body = json_body, encode = "json")
+  if (response$status != 200) {
+    log_error("Error posting Slack message")
+    stop()
+  }
+}
+
+#' Builds the header text, depending on how many signals are reported
+#'
+#' @param n_signals Number of signals reported
+#'
+#' @returns String header text
+slack_build_header <- function(n_signals) {
+  if (n_signals == 0) {
+    title <- paste0(Sys.Date(), ": No signals identified")
+  } else {
+    title <- paste0(":rotating_light: <!channel> ", Sys.Date(), ": ", n_signals, " alerts identified")
+  }
+  return(title)
+}
+
+#' Builds the signal alert text
+#'
+#' @param iso3 Country iso3 code
+#' @param indicator_id Indicator ID
+#' @param campaign_url URL of Mailchimp campaign to review
+#'
+#' @returns String signal alert text
+slack_build_alert <- function(iso3, indicator_id, campaign_url) {
+  return(paste0(iso3, ": ", indicator_id, " <", campaign_url, " | See draft campaign>\n"))
```

Comment:
Same on `return()`

Reply:
Also, depends on your style, but there are loads of options for string joining in R and this is one instance where you might prefer the `{glue}` style, which is basically like Python's f-strings.

```r
glue$glue(
  "{iso3}: {indicator_id} <{campaign_url} | See draft campaign>\n"
)
```

Reply:
Ooh yeah this is nicer! I played around with it, but seems to have some complexity with the escape characters, like `\n` -- will defs keep in mind for the future